quota/etcd/etcdqm/etcdqm.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package etcdqm contains an etcd-based quota.Manager implementation.
+package etcdqm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/quota/etcd/storage"
+)
+
+type manager struct {
+	qs *storage.QuotaStorage
+}
+
+// New returns a new etcd-based quota.Manager.
+func New(client *clientv3.Client) quota.Manager {
+	return &manager{qs: &storage.QuotaStorage{Client: client}}
+}
+
+func (m *manager) GetUser(ctx context.Context, req interface{}) string {
+	return "" // Unused
```

Comment:
That's the lever we can use for root-based quotas, for example. It's unused now, but it could easily come to life in the future.